tsconfig.dev.json

```diff
@@ -2,6 +2,8 @@
 	"extends": "@tsconfig/node22/tsconfig.json",
 	"compilerOptions": {
 		"noEmit": true,
+		"target": "es2022",
```

Comment:
Can you say more about this change / the need for it in the commit message?

I ask because the target is normally provided by our base tsconfig ([and appears to actually already be `es2022`](https://github.com/tsconfig/bases/blob/main/bases/node22.json)) -- which means that as we update that base this is going to eventually become a stale override, making it a non-ideal change.

Meanwhile the lib override removes a bunch of things that would otherwise be included.